File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdbool.h>

bool luhn(long duplicate);
void card_type(string number);

int main(void)
{
       long card=get_long("Card number:");
       char num[32];
       sprintf(num,"%ld",card);
       if (luhn(card))
       {
              card_type(num);
       }
       else
       {
              printf("INVALID\n");
       }
}

bool luhn(long duplicate)
{
       int counter=0;
       int product=0;
       int temp=0;
       while (duplicate)
       {
              if (counter%2!=0)
              {
                  temp=(duplicate%10)*2;
                  while (temp)
                  {
                     product+=temp%10;
                     temp/=10;
                  }
              }
              else
              {
                  product+=duplicate%10;
              }
              duplicate/=10;
              counter++;
       }
       return (product%10==0);
}

void card_type(string number)
{
       int len=strlen(number);
       if ( (len==15) && ( (number[0]=='3' && number[1]=='4')|| (number[0]=='3' && number[1]=='7')))
       {
              printf("AMEX\n");
       }
       else if ((len==13 || len==16) && number[0]=='4')
       {
              printf("VISA\n");
       }
       else if ((len)==16 && ( (number[0]=='5' && number[1]=='1') || ( number[0]=='5' && number[1]=='2') || (number[0]=='5' && number[1]=='3') || (number[0]=='5' && number[1]=='4') || (number[0]=='5' && number[1]=='5')))
       {
              printf("MASTERCARD\n");
       }
       else
       {
              printf("INVALID\n");
       }
}

File: Pset6/Credit/credit.py
import sys

number=None
while True:
      number = input("Number: ")
      if len(number)==0:
            print("Number must be entered!")
            continue

      if number.isnumeric():
            break
card = int(number)
add = 0
product = 0
count = 1
valid = False
while (card > 0):
      if count%2 == 0:
            temp = 2*(card % 10)
            while (temp > 0):
                  product+=temp%10
                  temp=temp//10
      else:
            add+=card%10
      count+=1
      card=card//10
sum=add+product
if sum%10==0:
     valid = True

if not valid:
      print("INVALID")
      sys.exit(1)
#count already done the work of the length
length=count-1
if ((length == 15) and ((int(number[0]+number[1])==34) or ((int(number[0]+number[1])==37)))):
      print("AMEX")
elif ((length==13 or length==16) and (int(number[0])==4)):
      print("VISA")
elif ((length==16) and (51<=int((number[0]+number[1]))<=55)):
      print("MASTERCARD")
else:
      print("No card matched")
      sys.exit(1)
sys.exit(0)






